server/cli/launch.py

```diff
@@ -135,6 +139,15 @@ def launch(
         file_parts = splitext(basename(data))
         title = file_parts[0]
 
+    if not fixed_port:
```

Comment:
if I am reading the code right, this will generate a warning in the case where the user has use defaults, and the default port is unavailable.

In this particular case,  where the user has let the system default, I'm not sure the warning adds any value or is all that friendly.  Consider suppressing it in this case...

Reply:
done